W-17615830-Setting-ACB-proxy-CP (#419)
mulesoft/docs-code-builder: modules/ROOT/pages/ref-proxy-settings.adoc

> = Configuring your Proxy in Anypoint Code Builder

Reviewer: ```suggestion
= Use a Proxy in Anypoint Code Builder
```
Not sure if this is accurate, but sounds to me that configuring is a subtask of using a proxy in/with ACB.

Author: I like your title! 
And as ACB titles are gerunds, it reads "Using a Proxy...".
Thanks!

:page-deployment-options: desktop-ide
:restart-ide: Restart the IDE to load all your changes. 

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

> Anypoint Code Builder supports proxy configuration through your current Visual Studio Code (VS Code) proxy settings. If your proxy settings are set correctly, the network-dependent functions, like publishing to Exchange, logging into Anypoint Platform, and running/debugging applications, work well through the proxy you set.

Reviewer: ```suggestion
Execute network-dependent functions, such as publishing to Exchange, logging into Anypoint Platform, and running/debugging applications, through a proxy by configuring your Visual Studio Code (VS Code) proxy settings.
```
User goal first. Confirm if it is accurate before applying.

Author: Yes! It's accurate. I just added "Anypoint Code Builder" to ensure the reader understand the goal of this article. Thanks!

[NOTE]
Proxy settings doesn’t work on the cloud IDE 

== Configure Your Proxy Settings in VS Code

When you configure your proxy setting in VS Code, you must take into account the following considerations:

> * When setting a proxy URL, the `protocol` can be `HTTP` or `SOCKS`. For example:

Reviewer 2: I think we should lowercase `HTTP` or `SOCKS` to `http` or `socks` because that is the format it will be in when they create their URL

Author: According to our style guide, it's "http" if it's part of a URL and "HTTP" if it's referring to the protocol.

Reviewer 2: right, in this case it is referring to the protocol in a URL, so it seems like either could apply. I just don't want users to be confused when they are creating their proxy URL and they put something like "HTTP://localhost:1080" when it should be "http://localhost:1080. what do you think?

Author: I can understand your concern. We should not receive complaints from customers about this topic. If so, we will review it if needed. Thanks!

> Configure the proxy settings through the VS Code settings UI. Anypoint Code Builder reads your settings from VS Code and puts them into its own tools and functions that need network connections so that requests for network connections go through the proxy you set up. When you configure your proxy setting in VS Code, you must take into account these considerations:

Reviewer 2: I personally still find it a bit confusing to give the extra details/notes on setting up before actually instructing them on how to set up. But I will defer to your judgement on this one!

Author: After you approve the PR, the editors should also approve it.
I'll share your point of view with them.
Thanks!

> Configure the proxy settings through the VS Code settings UI. Anypoint Code Builder reads your settings from VS Code and puts them into its own tools and functions that need network connections so that requests for network connections go through the proxy you set up. When you configure your proxy setting in VS Code, you must take into account these considerations:

Reviewer: ```suggestion
Anypoint Code Builder uses the configured proxy in your VS Code settings for tools and functions that need network connections.
```

I'd suggest adding this first sentence to the introduction topic. Then, add a 
**== Considerations** or **== Before you Begin Configuring your Proxy** section, including the next sentence and the rest:
_Consider the following when you configure your proxy setting in VS Code:_

* When setting a proxy URL...

Then:

**== Configure Your Proxy Settings in VS Code**

To configure proxy settings for Anypoint Code Builder:

. Open Visual Studio Code (VS Code)...

Author: I like it! Let's see if the new structure and content work well. Thanks!

** PROTOCOL://username:password@host:port
** PROTOCOL://host:port
+
[NOTE]
Anypoint Code Builder supports `SOCKS` proxies without authentication but doesn't support `SOCKS` proxies with basic authentication.
+

> * Anypoint Code Builder ignores these fields because it derives its settings from the proxy URL and the no-proxy list: 

Reviewer: ```suggestion
* Anypoint Code Builder ignores these fields and reads settings from the proxy URL and the no-proxy list: 
```
Apply if accurate.

Author: OK! Thanks!

** Proxy Authorization
** Proxy Kerberos Service Principal
** Proxy Strict SSL
+

> * If you must exclude certain hosts from the proxy, use the *Proxy bypass* settings.

Reviewer: ```suggestion
* To exclude certain hosts from the proxy, use the *Proxy bypass* settings.
```

Author: OK! Thanks!

> Configure the proxy settings through the VS Code settings UI. Anypoint Code Builder reads your settings from VS Code and puts them into its own tools and functions that need network connections so that requests for network connections go through the proxy you set up. For configuring your proxy settings:

Reviewer 2: Would it make more sense to put this section first explaining that they configure their settings through the VSCode UI, then put the section with the considerations after? It might be confusing to a use who is trying to set a proxy to see the considerations before knowing how they can even set their proxy settings. What do you think?

Reviewer 2: Meaning like put the section from lines 30-42 first, then the section from lines 14-28 after

Author: I made a change! Let's see if it works. Thanks!

. Open Visual Studio Code (VS Code).
. Open VS Code settings:

> .. In Windows, go to File > Preferences > Settings.
> .. In macOS, go to Code > Settings > Settings.

Reviewer: ```suggestion
.. In Windows, go to *File > Preferences > Settings*.
.. In macOS, go to *Code > Settings > Settings*.
```
These are UI elements right?

Author: Yes! They are UI elements. Great catch! Thanks!

. Search for "proxy" in the settings search bar.

> . Configure your proxy URL and no-proxy settings as needed.

Reviewer: ```suggestion
. Configure your proxy URL and no-proxy settings.
```

Author: OK! Thanks!

> . Configure your proxy settings in your Maven settings.xml file.

Reviewer 2: We can add this link here for users, showing them how to add their proxy to their maven settings.xml. 
https://maven.apache.org/guides/mini/guide-proxies.html

Author: Sure! Thanks for sharing this with me!

> . Configure your proxy settings in your Maven settings.xml file. See https://maven.apache.org/guides/mini/guide-proxies.html[Configure a Proxy in Maven].

Reviewer: ```suggestion
. Configure your proxy settings in your Maven `settings.xml` file. See https://maven.apache.org/guides/mini/guide-proxies.html[Configure a Proxy in Maven] for configuration instructions.
```

Author: OK! Thanks!

> . Restart VS Code to make the changes take effect.

Reviewer: ```suggestion
. Restart VS Code to apply your changes.
```

Author: OK! Thanks!

[NOTE]

> For Anypoint Code Builder to use proxy settings correctly, the 'Proxy Support' setting must be set to 'override' or 'on'.

Reviewer: _. Set the 'Proxy Support' setting to 'override' or 'on'._

This looks like an additional step to prevent issues. I'd add it to the list of steps to configure the proxy, as a mandatory step like the others, maybe just before the final Restart step.

Author: That's clever! Thanks!

== Add Anypoint Platform URLs to an Allowlist

> If you must add Anypoint Code Builder and Anypoint Platform URLs to an allowlist, see these tables:

Reviewer: ```suggestion
If your organization uses an allowlist, add the following Anypoint Code Builder and Anypoint Platform URLs:
```

Author: OK! Thanks!
